rai/ledger.cpp

```diff
@@ -495,12 +495,52 @@ bool rai::shared_ptr_block_hash::operator() (std::shared_ptr<rai::block> const &
 	return *lhs == *rhs;
 }
 
+rai::supply::supply (rai::ledger & ledger_a, rai::uint128_t const & inactive_supply_a) :
+ledger (ledger_a),
+inactive_supply (inactive_supply_a)
+{
+	circulating_update ();
+}
+
+rai::uint128_t rai::supply::circulating_get (bool force_update)
+{
+	rai::uint128_t supply;
+	{
+		std::unique_lock<std::mutex> lock (mutex);
+		if (force_update)
+		{
+			update_cache ();
+		}
+		supply = cached_supply;
+	}
+	return supply;
+}
+
+void rai::supply::circulating_update ()
+{
+	std::unique_lock<std::mutex> lock (mutex);
+	update_cache ();
+}
+
+void rai::supply::update_cache ()
+{
+	if (ledger.store.environment)
```

Comment:
This doesn't need to be checked, it should already be guaranteed.  Supply is a part of ledger and store appears before supply in the class definition so it will be initialized first.

Reply:
It's because of a check that fails if the check isn't there :( Not sure what the "right way" is.

Reply:
the test in question tests failure to open the db